api/v1/endpoints/listings.py

```diff
@@ -1,28 +1,81 @@
+from api.v1.serializers import list_of_listings
+from api.v1.parsers import endpoint_arguments, listing_arguments
 
-log = logging.getLogger(__name__)
+LOGGING_CONFIG = os.path.join(settings.ROOT_DIR, 'logging.conf')
+logging.config.fileConfig(LOGGING_CONFIG)
+log = logging.getLogger()
 
 ns = api.namespace('v1/listings', description='Manage FFA Listings in the Datatrust')
 
 
-@ns.route('/')
-class Listing(Resource):
-
-    @api.expect(pagination_arguments)
+@ns.route('/', methods=['GET', 'POST'])
+class Listings(Resource):
+    @api.expect(endpoint_arguments)
     @api.marshal_with(list_of_listings)
     def get(self):
         """
         Return a list of listings
         """
-        args = pagination_arguments.parse_args(request)
+        args = endpoint_arguments.parse_args(request)
         page = args.get('page', 1)
         per_page = args.get('per_page', 10)
 
         listings = ''
-        return listings
\ No newline at end of file
+        return listings
+
+@ns.route('/<listing_hash>', methods=['POST'])
+class Listing(Resource):
+    @api.expect(listing_arguments)
+    @api.response(201, 'Listing successfully added')
+    @api.response(500, 'Listing failed due to server side error')
+    def post(self, listing_hash):
+        """
+        Persist a new listing to file storage, db, and protocol
+        """
+        timings = {}
+        start_time = time.time()
+        file_type = None
+        md5_sum = None
+        tags = None
+        uploaded_md5 = None
+        if not request.form.get('file_type'):
+            return 'No file type specified', 400
+        else:
+            file_type = request.form.get('file_type')
+        if not request.form.get('md5_sum'):
+            return 'No md5 sum provided', 400
+        else:
+            md5_sum = request.form.get('md5_sum')
+        if request.form.get('tags'):
+            tags = request.form.get('tags')
+        for item in request.files.items():
+            destination = os.path.join('/tmp/uploads/')
+            log.info(f'Saving {item[0]} to {destination}')
+            if not os.path.exists(destination):
+                os.makedirs(destination)
+            item[1].save(f'{destination}{item[0]}')
+            with open(f'{destination}{item[0]}', 'rb') as data:
+                contents = data.read()
+                uploaded_md5 = hashlib.md5(contents).hexdigest()
+            if uploaded_md5 != md5_sum:
+                return 'File upload failed', 500
+            local_finish = time.time()
+            timings['local_save'] = local_finish - start_time
+            log.info(f'Saving {item[0]} to S3 bucket ffa-dev')
+            s3 = boto3.client('s3')
+            with open(f'{destination}{item[0]}', 'rb') as data:
+                s3.upload_fileobj(data, 'ffa-dev', item[0])
+            timings['s3_save'] = time.time() - local_finish
+            os.remove(f'{destination}{item[0]}')
+        log.info(timings)
+        return 'You uploaded a file'
```

Comment:
once we settle on some actual messaging should prob start putting them in a constants type file...